simple_node.go

```diff
@@ -84,6 +84,24 @@ func (node *SimpleNode) Equals(node2 Node) bool {
 	}
 
 	tag := node2.Tag()
+	//if both Ancestry sources, only check if their _APID is the same
+	if node.Tag().String() == "Source" && tag.String() == "Source" {
+		found := false
+	ancestry:
```

Comment:
Is this supposed to be below the loop? Otherwise a match would cause an infinite loop.

Reply:
What do you mean? Hand-checked this on my gedcom and it works great. Looking back, not sure why I didn't just return true in the loop.

Reply:
I want to pass a command line param to SimpleNode.Equals() to toggle on and off the ancestry source matching. How should I pass it to Equals(), other than flags.Lookup("ancestry-source-matching"), which indexes a map every time equals is called... (presumably bad performance)?